test/ClientCDFObjectTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Acquia\ContentHubClient\test;
+
+
+use Acquia\ContentHubClient\CDF\CDFObject;
+use Acquia\ContentHubClient\CDF\ClientCDFObject;
+use Acquia\ContentHubClient\CDFAttribute;
+use PHPUnit\Framework\TestCase;
+
+class ClientCDFObjectTest extends TestCase
+{
+  /**
+   * @var ClientCDFObject
+   */
+  private $clientCdfObject;
+
+  /**
+   *
+   */
+  public function setUp() : void
+  {
+    parent::setUp();
+    $settingsData = $this->getSettingsData();
+    try {
+      $this->clientCdfObject = ClientCDFObject::create(
+        'client_cdf_id_1',
+        $settingsData
+      );
+    } catch (\Exception $exception) {
+    }
+  }
+
+  /**
+   *
+   */
+  public function tearDown() :void
+  {
+    parent::tearDown();
+    unset($this->clientCdfObject);
+  }
+
+  /**
+   *
+   */
+  public function testGetClientName() : void
+  {
+    $this->assertInstanceOf(CDFAttribute::class, $this->clientCdfObject->getClientName());
+    $this->assertEquals(CDFAttribute::TYPE_STRING, $this->clientCdfObject->getClientName()->getType());
+    $this->assertEquals($this->getSettingsData()['settings']['name'], $this->clientCdfObject->getClientName()->getValue()[CDFObject::LANGUAGE_UNDETERMINED]);
```

Comment:
Use caching to prevent repetitive calls to this->clientCdfObject->getClientName().

Reply:
fixed